.github/workflows/_atlas-migrations-reusable.yml

```diff
@@ -0,0 +1,203 @@
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 30
+    outputs:
+      status: ${{ steps.migration-result.outputs.status }}
+      applied_count: ${{ steps.migration-result.outputs.applied_count }}
+    
+    services:
+      postgres:
+        image: ${{ inputs.postgres_image }}
+        env:
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: test_db
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ inputs.go_version }}
+
+      - name: Install Atlas
+        run: |
+          curl -sSf https://atlasgo.sh | sh -s -- --version ${{ inputs.atlas_version }}
+
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download Go dependencies
+        run: go mod download
+
+      - name: Install PostgreSQL client
+        if: inputs.action == 'validate'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql-client
+
+      - name: Clean PostGIS schemas for Atlas
+        if: inputs.action == 'validate'
+        run: |
+          # Remove PostGIS-created schemas that interfere with Atlas clean database requirement
+          PGPASSWORD=postgres psql -h localhost -U postgres -d test_db -c "
+            DROP SCHEMA IF EXISTS tiger CASCADE;
+            DROP SCHEMA IF EXISTS tiger_data CASCADE;
+            DROP SCHEMA IF EXISTS topology CASCADE;
+          "
+
+      - name: Validate migration directory
+        if: inputs.action == 'validate' || inputs.action == 'deploy'
+        run: |
+          atlas migrate validate \
+            --dir "file://${{ inputs.migrations_path }}" \
+            --dev-url "docker+postgres://_/${{ inputs.postgres_image }}/dev?search_path=public"
+
+      - name: Lint migrations
+        if: inputs.action == 'validate' || inputs.action == 'deploy'
+        run: |
+          atlas migrate lint \
+            --dir "file://${{ inputs.migrations_path }}" \
+            --dev-url "docker+postgres://_/${{ inputs.postgres_image }}/dev?search_path=public" \
+            --latest 1
+
+      - name: Test migration up/down
+        if: inputs.action == 'validate'
+        run: |
+          # Test applying migrations to the service database
+          # For testing on clean database, we apply all migrations from the beginning
+          atlas migrate apply \
+            --dir "file://${{ inputs.migrations_path }}" \
+            --url "postgres://postgres:postgres@localhost:5432/test_db?sslmode=disable"
+          
+          echo "✅ Migrations applied successfully"
```

Comment:
Shouldn't we validate before applying directly? I know it is a testnet db, but maybe the output is more informative than a failure on the test db directly